src/ray/core_worker/transport/raylet_transport.cc

```diff
@@ -29,4 +28,28 @@ Status CoreWorkerRayletTaskReceiver::GetTasks(std::vector<TaskSpec> *tasks) {
   return Status::OK();
 }
 
+CoreWorkerRayletTaskReceiver::CoreWorkerRayletTaskReceiver(
+    std::unique_ptr<RayletClient> &raylet_client, boost::asio::io_service &io_service,
+    rpc::GrpcServer &server)
+    : raylet_client_(raylet_client), task_service_(io_service, *this) {
+  server.RegisterService(task_service_);
+}
+
+void CoreWorkerRayletTaskReceiver::HandleAssignTask(
+    const rpc::AssignTaskRequest &request, rpc::AssignTaskReply *reply,
+    rpc::RequestDoneCallback done_callback) {
+  const std::string &task_message = request.task_spec();
+  const raylet::Task task(*flatbuffers::GetRoot<protocol::Task>(
+      reinterpret_cast<const uint8_t *>(task_message.data())));
```

Comment:
`static_cast` instead of `reinterpret_cast`?

Reply:
that doesn't work unfortunately. see https://stackoverflow.com/questions/10151834/why-cant-i-static-cast-between-char-and-unsigned-char